source/common/http/conn_manager_impl.cc

```diff
@@ -905,8 +905,10 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
       request_headers_->Expect()->value() == Headers::get().ExpectValues._100Continue.c_str()) {
     // Note in the case Envoy is handling 100-Continue complexity, it skips the filter chain
     // and sends the 100-Continue directly to the encoder.
-    chargeStats(continueHeader());
-    response_encoder_->encode100ContinueHeaders(continueHeader());
+    const auto& continue_headers = continueHeader();
+    const auto status = response_encoder_->encode100ContinueHeaders(continue_headers);
+    ASSERT(status.ok());
```

Comment:
Could you add a very quick comment that this should always hold because `continueHeader()` has a valid status?

(This could also be a RELEASE_ASSERT? If the error status ever got more restrictive it would be pretty easily detectable since continueHeader() is hard-coded)

Reply:
done! Thanks!